### DISCUSSION TITLE
Tracing a request across rabbitmq

### DISCUSSION THREAD

**Discussion by soumit09** [open-telemetry/opentelemetry-java-instrumentation#3412] [Q&A]
I am using auto instrument java agent. I can see my rest api calls ends tracing when it sends message to the queue. Is there any option where same trace can be seen across sends message to rabbitmq and then listener picks it and do other stuffs? 
otel.agent.version =1.1.0

**Comment by mateuszrzeszutek** [comment]
Hey @soumit09 ,

Are you using the spring-rabbit (or spring-cloud-stream-rabbitmq-binder) library? If so, please try using the latest javaagent release [1.3.1](https://github.com/open-telemetry/opentelemetry-java-instrumentation/releases/tag/v1.3.1), we've had a couple of fixes related to rabbitMq context propagation since 1.1.0 (https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3339, https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3295, https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3381)

**Reply by soumit09** [reply]
Yes i am using spring-rabbit. Let me try new version. Thanks for quick reply.

**Reply by soumit09** [reply]
It working now. I can see same trace across rabbitmq sender and listener.
